### PR TITLE
[POC] Adding timeout-based retries to DAG transfers

### PR DESCRIPTION
**Pull request by plauche**
The bulk of this PR was adding `myceli/src/shipper.rs` which consolidates the data exchange code (from `transmit.rs` and `receiver.rs`) and adds enough state to support timeout-based retries when transmitting DAGs.

### Added

- Added retries to the `TransmitDag` API and timeout-based mechanism for retrying dag transmission, this included adding retry duration as a CLI arg to `myceli`.
- Added protocol messages for requesting missing dag blocks, requesting transmission of dags & blocks, and retrying dag transmission.
- Added a basic Dockerfile for building and running `myceli`.
- Added `shipper` struct to `myceli` which consolidates all data exchange code and adds state necessary to support timeout-based retries during DAG transmission.

### Changed

- Reworked `myceli` to operate in the traditional multi-threaded fashion instead of using async. This included reworking `myceli::Listener` to spawn `shipper` in a separate thread and pass along messages via channel.
- Updated logging in the desktop `radio-service` to use `tracing`.
- Moved `transmit_dag` and `transmit_block` functionality from handlers to `shipper`.

### Removed

- Removed the `TransmitFile` API, all files to be transmitted should be imported and then transmitted to support retransmission of blocks.

### CODE REVIEW

File: local-dev-environment/desktop/radio-service/Cargo.toml [hunk 1]
```diff
@@ -8,4 +8,6 @@ edition = "2021"
 [dependencies]
 clap = { version = "4.0", features = ["derive"] }
 tokio = { version = "1.8", features = ["macros", "rt", "rt-multi-thread"] }
```
Comment (ddimaria) [inline local-dev-environment/desktop/radio-service/Cargo.toml:10]: Now that you're moving off of an async runtime, does it make sense to use the std UdpSocket and remove all of tokio?

File: local-storage/src/storage.rs [hunk 2]
```diff
@@ -19,13 +19,18 @@ impl Storage {
...
 
-    pub async fn import_path(&self, path: &Path) -> Result<String> {
-        let file: File = FileBuilder::new()
-            .path(path)
-            .fixed_chunker(50)
-            .build()
-            .await?;
-        let blocks: Vec<_> = file.encode().await?.try_collect().await?;
+    pub fn import_path(&self, path: &Path) -> Result<String> {
+        let rt = tokio::runtime::Runtime::new().unwrap();
```
Comment (ddimaria) [inline local-storage/src/storage.rs:23]: Do you still need this runtime?

File: local-storage/src/storage.rs [hunk 3]
```diff
@@ -19,13 +19,18 @@ impl Storage {
...
+        let blocks = rt.block_on(async {
+            let file: File = FileBuilder::new()
+                .path(path)
+                .fixed_chunker(50)
+                .build()
+                .await
+                .unwrap();
```
Comment (ddimaria) [inline local-storage/src/storage.rs:30]: Unwrap here and below

File: myceli.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Base build stage
```
Comment (ddimaria) [inline myceli.Dockerfile:1]: I didn't see at first pass, but can you document how you are invoking this image in the README?  We could setup a docker-compose network and have everything run on it locally, as well as a similar setup for CI tests.

File: local-dev-environment/desktop/radio-service/Cargo.toml (hunk 1 above)
Comment (plauche) [reply local-dev-environment/desktop/radio-service/Cargo.toml:10]: I didn't actually un-async this service in this PR, so the tokio dependencies will stay for now in this one.

**Pull request merged**